internal/controller/runtime/fsm/runtime_fsm_select_shoot_processing.go

```diff
@@ -19,38 +19,42 @@ func sFnSelectShootProcessing(_ context.Context, m *fsm, s *systemState) (stateF
 	if s.shoot.Spec.DNS == nil || s.shoot.Spec.DNS.Domain == nil {
 		msg := fmt.Sprintf("DNS Domain is not set yet for shoot: %s, scheduling for retry", s.shoot.Name)
 		m.log.Info(msg)
-		return updateStatusAndRequeueAfter(m.RCCfg.GardenerRequeueDuration)
+		return requeueAfter(m.RCCfg.GardenerRequeueDuration)
 	}
 
 	lastOperation := s.shoot.Status.LastOperation
 	if lastOperation == nil {
 		msg := fmt.Sprintf("Last operation is nil for shoot: %s, scheduling for retry", s.shoot.Name)
 		m.log.Info(msg)
-		return updateStatusAndRequeueAfter(m.RCCfg.GardenerRequeueDuration)
+		return requeueAfter(m.RCCfg.GardenerRequeueDuration)
 	}
 
 	patchShoot, err := shouldPatchShoot(&s.instance, s.shoot)
 	if err != nil {
 		msg := fmt.Sprintf("Failed to get applied generation for shoot: %s, scheduling for retry", s.shoot.Name)
 		m.log.Error(err, msg)
-		return updateStatusAndStop()
+		return requeueAfter(m.RCCfg.GardenerRequeueDuration)
 	}
 
 	if patchShoot {
 		m.log.Info("Gardener shoot already exists, updating")
 		return switchState(sFnPatchExistingShoot)
 	}
 
-	if lastOperation.Type == gardener.LastOperationTypeCreate {
-		return switchState(sFnWaitForShootCreation)
-	}
+	if s.instance.Status.State == imv1.RuntimeStatePending || s.instance.Status.State == "" {
```

Comment:
Is it possible that something will cause here a `Status.State == Error` which we would like to requeue/heal from? 
I will try to simulate and check that.

Reply:
This will be hopefully handled by this code above

``` golang
	patchShoot, err := shouldPatchShoot(&s.instance, s.shoot)
	if err != nil {
		msg := fmt.Sprintf("Failed to get applied generation for shoot: %s, scheduling for retry", s.shoot.Name)
		m.log.Error(err, msg)
		return requeueAfter(m.RCCfg.GardenerRequeueDuration)
	}

	if patchShoot {
		m.log.Info("Gardener shoot already exists, updating")
		return switchState(sFnPatchExistingShoot)
	}
```